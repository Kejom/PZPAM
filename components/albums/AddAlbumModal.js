import { Modal, StyleSheet, View, ToastAndroid } from "react-native";
import Input from "../shared/Input";
import { useState } from "react";
import Button from "../shared/Button";
import { useDispatch, useSelector } from "react-redux";
import { addAlbum } from "../../redux/albums";
import { GlobalStyles } from "../../constants/style";

export default function AddAlbumModal({ isVisible, onClose }) {
    const dispatch = useDispatch();
    const currentUserId = useSelector(state => state.users.loggedUserId);
    const [albumTitle, setAlbumTitle] = useState("");

    function onCreatePress(){
        if(!albumTitle)
            return ToastAndroid.showWithGravity("Tytuł albumu nie może być pusty", ToastAndroid.LONG, ToastAndroid.TOP);
        
        const newAlbum = {userId: currentUserId, title: albumTitle};
        dispatch(addAlbum(newAlbum));
        onClose();
    } 

    return (
        <Modal
            animationType="slide"
            transparent={true}
            visible={isVisible}
            onRequestClose={onClose}>
            <View style={GlobalStyles.defaultModalContainer}>
                <Input
                    label="Tytuł Albumu:"
                    isValid={true}
                    textInputConfig={{
                        inputMode: 'text',
                        onChangeText: setAlbumTitle,
                        value: albumTitle
                    }} />
                    <View style={styles.buttonsRow}>
                    <Button style={styles.button} onPress={onCreatePress}>Stwórz Album</Button>
                    <Button style={styles.button} onPress={onClose}>Anuluj</Button>
                    </View>

            </View>
        </Modal>
    )
}

styles = StyleSheet.create({
    buttonsRow: {
        flexDirection: 'row',

    },
    button: {
        flex: 1,
        marginHorizontal: 10
    }
})