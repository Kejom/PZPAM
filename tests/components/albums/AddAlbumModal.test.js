import React from 'react';
import { render} from '@testing-library/react-native';
import AddAlbumModal from '../../../components/albums/AddAlbumModal';
import { Provider } from 'react-redux';
import configureStore from "redux-mock-store"

describe('AddAlbumModal', () => {
    const mockStore = configureStore([]);
    const store = mockStore({
        users: {
          loggedUserId: '123',
        },
      });
  
    it('renders correctly', () => {
      const { toJSON } = render(
        <Provider store={store}>
          <AddAlbumModal isVisible={true} onClose={() => {}} />
        </Provider>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });