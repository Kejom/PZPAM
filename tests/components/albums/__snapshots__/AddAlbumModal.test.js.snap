// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAlbumModal renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "backgroundColor": "#4CAF50",
        "borderColor": "#2B702D",
        "borderRadius": 8,
        "borderWidth": 1,
        "elevation": 8,
        "margin": 10,
        "marginTop": 100,
        "padding": 16,
      }
    }
  >
    <View
      style={
        [
          {
            "marginHorizontal": 4,
            "marginVertical": 8,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#58FC5F",
              "fontSize": 12,
              "marginBottom": 4,
            },
            false,
          ]
        }
      >
        Tytuł Albumu:
      </Text>
      <TextInput
        inputMode="text"
        onChangeText={[Function]}
        style={
          [
            {
              "backgroundColor": "#58FC5F",
              "borderRadius": 6,
              "color": "#2B702D",
              "fontSize": 18,
              "padding": 6,
            },
            false,
            undefined,
          ]
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "marginHorizontal": 10,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={false}
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#2B702D",
                  "borderRadius": 4,
                  "padding": 8,
                },
                false,
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "white",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Stwórz Album
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
            "marginHorizontal": 10,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={false}
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#2B702D",
                  "borderRadius": 4,
                  "padding": 8,
                },
                false,
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "white",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Anuluj
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
